### PR TITLE
WDL-9: GCS upload

### PR DESCRIPTION
**Pull request by JiaweiChen110**
As a user of WDL, I want to be able to use files in GCS buckets, so we keep data in a consistent location

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 TARGET = install
-VERSION=1.3.0
+VERSION=1.3.0-jchen-upload
```
Comment (JiaweiChen110) [inline Makefile]: Change to default

File: src/main/python/gcp/gcs.py [hunk 2]
```diff
@@ -166,6 +168,37 @@ def download(config: DownloadConfig):
...
+    """
+    client = storage.Client()
+    
+    """Uploads a file to the bucket."""
+    # The ID of your GCS bucket
+    # sourceBucket = "your-bucket-name"
+    # The path to your file to upload
+    # sourceFile = "local/path/to/file"
+    # The ID of your GCS object
+    # destination_uri = "storage-object-name"
```
Comment (JiaweiChen110) [inline src/main/python/gcp/gcs.py]: change destination_uri to sourcePrefix

File: src/main/wdl/gcs.wdl [hunk 3]
```diff
@@ -38,7 +38,7 @@ task Compose {
 
       Int cpu = 1
       String memory = "128 MB"
-      String dockerImage = "wdl-kit:1.3.0"
+      String dockerImage = "wdl-kit:1.3.0-jchen-upload"
```
Comment (JiaweiChen110) [inline src/main/wdl/gcs.wdl]: change dockerimage back to default

File: src/main/wdl/gcs.wdl [hunk 4]
```diff
@@ -125,3 +125,55 @@ task Download {
...
     }
 }
+
+struct UploadConfig {
+  # Source bucket
+  String sourceBucket
+  # prefix used to find source blobs in bucket
+  String sourcePrefix
+  # Source file directory
+  String sourceFile
```
Comment (JiaweiChen110) [inline src/main/wdl/gcs.wdl]: change type String to File

File: src/main/wdl/gcs.wdl [hunk 5]
```diff
@@ -125,3 +125,55 @@ task Download {
       memory: memory
     }
 }
+
+struct UploadConfig {
+  # Source bucket
+  String sourceBucket
+  # prefix used to find source blobs in bucket
+  String sourcePrefix
+  # Source file directory
```
Comment (JiaweiChen110) [inline src/main/wdl/gcs.wdl]: change directory to file

File: src/main/wdl/gcs.wdl [hunk 6]
```diff
@@ -125,3 +125,55 @@ task Download {
...
+
+    input {
+      File? credentials
+      String? projectId
+      String sourceBucket
+      String sourcePrefix
+      File sourceFile
+      Int cpu = 1
+      String memory = "128 MB"
+      String dockerImage = "wdl-kit:1.3.0-jchen-upload"
```
Comment (JiaweiChen110) [inline src/main/wdl/gcs.wdl]: change dockerimage back to default

File: tests/input-gcs.json [hunk 7]
```diff
@@ -1,8 +1,11 @@
 {
-    "IntegrationTestGcs.testCredentials": "credentials.json",
-    "IntegrationTestGcs.testProjectId": "xxxxxxxxx",
-    "IntegrationTestGcs.gcsDestination": "gs://xxxxxx_test/loadfiletest/gcs_result.txt",
+    "IntegrationTestGcs.testCredentials": "/home/jchen313_stanford_edu/cwl/som-rit-phi-starr-dev-key.json",
```
Comment (JiaweiChen110) [inline tests/input-gcs.json]: Change all these back to default

File: tests/integration-test-gcs.wdl [hunk 8]
```diff
@@ -84,7 +99,7 @@ workflow IntegrationTestGcs {
         String? testGcsCompose = if defined(TEST_GCS_COMPOSE) then 'Passed' else 'Failed'
         String? testGcsDownload = if defined(TEST_GCS_DOWNLOAD) then 'Passed' else 'Failed'   
         String? finalTestResult = finalResult
-        #String? downloadFile = ShowSourceFile.fileContent
```
Comment (JiaweiChen110) [inline tests/integration-test-gcs.wdl:87]: check TEST_GCS_UPLOAD if true for pass else fail

**Review by JiaweiChen110 — COMMENTED** [review]
Made all the fixes

**Review by jmesterh — APPROVED** [review]

**Pull request merged**